core/lib/assert.c

```diff
@@ -24,29 +24,34 @@
 #include "backtrace.h"
 #endif
 
-__NORETURN void _assert_failure(const char *file, unsigned line)
+__NORETURN static inline void _assert_common(void)
 {
-    printf("%s:%u => ", file, line);
 #if IS_USED(MODULE_BACKTRACE)
+#ifdef DEBUG_ASSERT_VERBOSE
     printf("failed assertion. Backtrace:\n");
+#endif
     backtrace_print();
 #endif
 #ifdef DEBUG_ASSERT_BREAKPOINT
     DEBUG_BREAKPOINT(1);
 #endif
+    if (DEBUG_ASSERT_ZOMBIFY) {
+        puts("FAILED ASSERTION.");
+        thread_zombify();
```

Comment:
Is the intention of zombification that a zombie thread can just be reaped when another thread detects that it is a zombie, or would it need some confirmation that the thread is now completed and then zombiefies itself before it can legitimately reap it?

Thing is: If a thread reaches an assert and thus diverges, we can not allow another thread to be spawned in its place, for this thread may have allowed other parts to use its stack, so that reaping can not be allowed.

I originally suggested `while (true) { thread_sleep(); }`, this would be sure to not be reused.